### PR TITLE
Use "you" instead of your display name for notice events you have sent

### PR DESCRIPTION
**Pull request by bmarty**
It's worth noting that historically notice string were located in the SDK. Now a few of them are in the app.
This PR does not change anything about this, but maybe on day we should clarify this point.

### CODE REVIEW

File: matrix-sdk-android/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2,53 +2,85 @@
 <resources>
     <string name="summary_message">%1$s: %2$s</string>
     <string name="summary_user_sent_image">%1$s sent an image.</string>
+    <string name="summary_you_sent_image">You sent an image.</string>
```
Comment (onurays) [inline matrix-sdk-android/src/main/res/values/strings.xml:5]: Good idea to use new strings for the sake of localization.

File: matrix-sdk-android/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2,53 +2,85 @@
...
     <string name="notice_room_ban">%1$s banned %2$s</string>
+    <string name="notice_room_ban_by_you">You banned %1$s</string>
     <string name="notice_room_withdraw">%1$s withdrew %2$s\'s invitation</string>
+    <string name="notice_room_withdraw_by_you">You withdrew %1$s\'s invitation</string>
     <string name="notice_avatar_url_changed">%1$s changed their avatar</string>
+    <string name="notice_avatar_url_changed_by_you">You changed your avatar</string>
     <string name="notice_display_name_set">%1$s set their display name to %2$s</string>
+    <string name="notice_display_name_set_by_you">You set your display name to %1$s</string>
     <string name="notice_display_name_changed_from">%1$s changed their display name from %2$s to %3$s</string>
+    <string name="notice_display_name_changed_from_by_you">You changed their display name from %1$s to %2$s</string>
```
Comment (onurays) [inline matrix-sdk-android/src/main/res/values/strings.xml:35]: Should be:
You changed **your** display name from %1$s to %2$s

File: matrix-sdk-android/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -2,53 +2,85 @@
...
     <string name="notice_room_name_removed">%1$s removed the room name</string>
+    <string name="notice_room_name_removed_by_you">You removed the room name</string>
     <string name="notice_room_topic_removed">%1$s removed the room topic</string>
+    <string name="notice_room_topic_removed_by_you">You removed the room topic</string>
     <string name="notice_event_redacted">Message removed</string>
     <string name="notice_event_redacted_by">Message removed by %1$s</string>
     <string name="notice_event_redacted_with_reason">Message removed [reason: %1$s]</string>
     <string name="notice_event_redacted_by_with_reason">Message removed by %1$s [reason: %2$s]</string>
     <string name="notice_profile_change_redacted">%1$s updated their profile %2$s</string>
+    <string name="notice_profile_change_redacted_by_you">You updated your profile %2$s</string>
```
Comment (onurays) [inline matrix-sdk-android/src/main/res/values/strings.xml:77]: Should be %1$s.

File: matrix-sdk-android/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -245,39 +277,68 @@
     <string name="clear_timeline_send_queue">Clear sending queue</string>
 
     <string name="notice_room_invite_no_invitee_with_reason">%1$s\'s invitation. Reason: %2$s</string>
+    <string name="notice_room_invite_no_invitee_with_reason_by_you">Your invitation. Reason: %2$s</string>
```
Comment (onurays) [inline matrix-sdk-android/src/main/res/values/strings.xml:280]: Should be %1$s.

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt [hunk 5]
```diff
@@ -103,28 +105,54 @@ class NoticeEventFormatter @Inject constructor(private val sessionHolder: Active
...
+                        sp.getString(R.string.notice_room_created, it.creator)
+                    }
+                }
     }
 
     private fun formatRoomNameEvent(event: Event, senderName: String?): CharSequence? {
         val content = event.getClearContent().toModel<RoomNameContent>() ?: return null
         return if (content.name.isNullOrBlank()) {
-            sp.getString(R.string.notice_room_name_removed, senderName)
+            if (event.isSentByCurrentUser()) {
```
Comment (onurays) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:120]: I didn't like these if-else statements, but don't have a better approach at the moment.

File: matrix-sdk-android/src/main/res/values/strings.xml (hunk 2 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/res/values/strings.xml:35]: good catch, thanks!

File: matrix-sdk-android/src/main/res/values/strings.xml (hunk 3 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/res/values/strings.xml:77]: good catch, thanks!

File: matrix-sdk-android/src/main/res/values/strings.xml (hunk 4 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/res/values/strings.xml:280]: good catch, thanks!

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt (hunk 5 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/format/NoticeEventFormatter.kt:120]: Yes, it's not ideal...

**Pull request merged**